packages/@aws-cdk/aws-codepipeline-actions/test/s3/test.s3-cache-control.ts

```diff
@@ -0,0 +1,36 @@
+import { Duration } from "@aws-cdk/core";
+import { Test } from 'nodeunit';
+import cpactions = require('../../lib');
+
+// tslint:disable:object-literal-key-quotes
+
+const testDuration = Duration.hours(1);
+const serialize = cpactions.serializeCacheControlOptions;
+
+export = {
+  'individual options'(test: Test) {
+    test.equal(serialize({maxAge: testDuration}), 'max-age=3600');
+    test.equal(serialize({noCache: true}), 'no-cache');
+    test.equal(serialize({noStore: true}), 'no-store');
+    test.equal(serialize({noTransform: true}), 'no-transform');
+    test.equal(serialize({of: ['arbitrary!Value']}), 'arbitrary!Value');
+    test.equal(serialize({mustRevalidate: true}), 'must-revalidate');
+    test.equal(serialize({public: true}), 'public');
+    test.equal(serialize({private: true}), 'private');
+    test.equal(serialize({proxyRevalidate: true}), 'proxy-revalidate');
+    test.equal(serialize({sMaxAge: testDuration}), 's-maxage=3600');
+    test.done();
+  },
+  'falsy value'(test: Test) {
+    test.equal(serialize({}), '');
```

Comment:
Is an empty string a legal RFC-2616 header? If not, we should throw an exception here.

Reply:
You're right, it is illegal to have an empty Cache-Control value:
```
    Cache-Control   = "Cache-Control" ":" 1#cache-directive
```

I've added an exception to handle it, thanks